zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -200,6 +200,110 @@ impl<F: FieldExt, const N: usize> AddWordsGadget<F, N> {
     }
 }
 
+/// Construction of 256-bit product by 256-bit multiplicand * 64-bit multiplier.
+#[derive(Clone, Debug)]
+pub(crate) struct MulWordByU64Gadget<F> {
```

Comment:
There's a duplicate Multiply gadget in #255 . Probably @han0110 you can also take a look

Reply:
@noel2004 Please also take a look at this part.

Reply:
Seems this Gadget is specialized for the mutiply of `U256 * U64 mod 2^256` while our gadget can be applied for two full (U)256 integer (the I256 can also be applicable). So maybe they are for different usage caes ...

I just wonder is it a good designation to keep the constraint of mutiplier outside of the gadget？

Reply:
yes this is specialized for multiplication of `U256 * U64 mod 2^256`, and I think the one in #255 is more general, maybe we can extend the one in #255 to support such special case, which would not carry overflow 64-bit and could save 2 cells.

If #255 is merged first, I can do so, or we can also leave that for further PRs.

Reply:
Does it save constraints the fact of having the specialised gadget? 

IF so, we could maybe keep both. And use the U64 in the cases where we know the input has to be below 64bits.

WDYT?

Reply:
It indeed saves some, but even with same struct, I think we can still enforce different constraint by some const generic flag. But not sure if it's worthy to add such complexity to avoid re-implementation on similar logic.

Reply:
Well if it only depends on adding const-generics I think we're fine. 

The only thing I see is that the const_generics should not be `ANY` value and just `64` or `256`.
We can maybe file an issue to discuss about it.